src/System.Management.Automation/engine/CommandSearcher.cs

```diff
@@ -978,6 +988,14 @@ private CmdletInfo GetNextCmdlet()
                                     matchingCmdletInfo.Add(cmdlet);
                                 }
                             }
+                            else if (_commandResolutionOptions.HasFlag(SearchResolutionOptions.UseAbbreviationExpansion))
+                            {
+                                string abbreviatedCmdlet = new string(cmdlet.Name.Where(c => char.IsUpper(c) || c == '-').ToArray());
+                                if (_commandName.Equals(abbreviatedCmdlet, StringComparison.OrdinalIgnoreCase))
```

Comment:
I wonder if this should be `abbreviatedCmdlet.IndexOf(_commandName, StringComparison.OrdinalIgnoreCase) == 0`
which would allow `pschr` to match `PSConsoleHostReadLine`

Reply:
The current code already resolves `pschr` as `PSConsoleHostReadLine`.  That line is so that `i-psdf` resolves `Import-PowerShellDataFile`

Reply:
see my response below